src/quick-lint-js/parse.h

```diff
@@ -1290,12 +1287,32 @@ class parser {
       QLJS_PARSER_UNIMPLEMENTED();
       break;
     }
-
+  }
+  template <QLJS_PARSE_VISITOR Visitor>
+  void parse_and_visit_function_body_no_scope(Visitor &v) {
     v.visit_enter_function_scope_body();
-
     this->parse_and_visit_statement_block_no_scope(v);
   }
 
+  template <QLJS_PARSE_VISITOR Visitor>
+  void parse_and_visit_function_parameters_and_body_no_scope(
+      Visitor &v, std::optional<source_code_span> name,
+      function_attributes attributes) {
+    function_guard guard = this->enter_function(attributes);
+    this->parse_and_visit_function_parameters_no_scope(v, name);
+    // function f() // Invalid.
+    if (this->peek().type != token_type::left_curly) {
+      if (!name.has_value()) {
+        QLJS_PARSER_UNIMPLEMENTED();
+      }
+      this->error_reporter_->report(error_missing_function_body{
+          .function_name = *name,
+      });
```

Comment:
I think it would be better to show the error after the parameter list's `)`, similar to E064 (*missing body for 'if' statement*):
```suggestion
      this->error_reporter_->report(error_missing_function_body{
          .expected_body = this->lexer_.end_of_previous_token(),
      });
```